extension/parquet/parquet_extension.cpp

```diff
@@ -492,7 +492,16 @@ class ParquetScanFunction {
 					// The current reader has rowgroups left to be scanned
 					scan_data.reader = parallel_state.readers[parallel_state.file_index];
 					vector<idx_t> group_indexes {parallel_state.row_group_index};
-					scan_data.reader->InitializeScan(scan_data.scan_state, group_indexes);
+					unordered_set<string> used_columns;
+					// create set to keep track of which columns are used
+					for (auto col_idx : parallel_state.column_ids) {
+						// col_idx could be -1
+						if (IsRowIdColumnId(col_idx)) {
+							continue;
+						}
+						used_columns.insert(StringUtil::Lower(bind_data.names[col_idx]));
```

Comment:
This `Lower` call would also not be necessary if we use a case-insensitive set instead